source/Projectile.cpp

```diff
@@ -34,24 +34,17 @@ namespace {
 }
 
 
-
-Projectile::Projectile(const Ship &parent, Point position, Angle angle, const Weapon *weapon)
-	: Body(weapon->WeaponSprite(), position, parent.Velocity(), angle),
-	weapon(weapon), targetShip(parent.GetTargetShip()), lifetime(weapon->Lifetime())
+//contains statement needed in multiple constructors.
+void Projectile::Init(const Government *gov, const Weapon *weapon)
```

Comment:
Methods are to be defined in the order of their declaration. Since the private methods are declared after the public ones in the header, so too will its definition be after the public methods. You declared `Init` before `CheckLock`, so it should be before `CheckLock`.